File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/NewsListModule.java
package com.androidacademy.msk.exerciseproject.di.module;

import android.support.annotation.NonNull;

import com.androidacademy.msk.exerciseproject.data.database.dao.NewsDao;
import com.androidacademy.msk.exerciseproject.data.network.api.NYTimesApi;
import com.androidacademy.msk.exerciseproject.di.scope.FragmentScope;
import com.androidacademy.msk.exerciseproject.screen.news_list.NewsListPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class NewsListModule {

    @Provides
    @FragmentScope
    @NonNull
    public NewsListPresenter providePresenter(@NonNull NYTimesApi api, @NonNull NewsDao dao) {
        return new NewsListPresenter(api, dao);
    }
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/NewsListModule.kt
package com.androidacademy.msk.exerciseproject.di.module

import com.androidacademy.msk.exerciseproject.screen.newslist.NewsListPresenter
import toothpick.config.Module


class NewsListModule : Module() {
    init {
        bind(NewsListPresenter::class.java).singletonInScope()
    }
}